pkg/java/pom/testdata/repository/org/example/example-child-parent/2.0.0/example-child-parent-2.0.0.pom

```diff
@@ -0,0 +1,23 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```

Comment:
I took a read at the doc again.
https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#dependency-management

I found the following statement.

>Finally, since d is specified in B's dependency management section, should d be a dependency (or transitive dependency) of a or c, version 1.0 will be chosen - again because dependency management takes precedence over dependency mediation and also because the current POM's declaration takes precedence over its parent's declaration.

Another question came to my mind. You added 4 poms for testing. If all of them have `dependencyManagement`, what is the priority on `org.example:example-child:2.0.0`? According to the above statement, `dependencyManagement` in `org.example:example-child:2.0.0` is the highest priority. What about others?

1. org.example:example-child:2.0.0 (repository/org/example/example-child/2.0.0/example-child-2.0.0.pom)
2. com.example:parent:3.0.0 (transitive-parents/pom.xml)
3. com.example:base:4.0.0 (transitive-parents/base/pom.xml)
4. org.example:example-child-parent:2.0.0 (repository/org/example/example-child/2.0.0/example-child-parent-2.0.0.pom)

or

1. org.example:example-child:2.0.0 (repository/org/example/example-child/2.0.0/example-child-2.0.0.pom)
2. com.example:base:4.0.0 (transitive-parents/base/pom.xml)
3. com.example:parent:3.0.0 (transitive-parents/pom.xml)
4. org.example:example-child-parent:2.0.0 (repository/org/example/example-child/2.0.0/example-child-parent-2.0.0.pom)

or

1. org.example:example-child:2.0.0 (repository/org/example/example-child/2.0.0/example-child-2.0.0.pom)
2. com.example:parent:3.0.0 (transitive-parents/pom.xml)
3. org.example:example-child-parent:2.0.0 (repository/org/example/example-child/2.0.0/example-child-parent-2.0.0.pom)
4. com.example:base:4.0.0 (transitive-parents/base/pom.xml)

or something else? In your current example, only parents (`com.example:parent:3.0.0` and `org.example:example-child-parent:2.0.0` ) have `dependencyManagement`, but what if children also have `dependencyManagement`?

Reply:
To clarify my question, I drew a diagram. The version of `jackson-databind` should be `3.0.0`, but what is next if `3.0.0` is not specified? `1.0.0` or `2.0.0`?

![image](https://user-images.githubusercontent.com/2253692/185158754-d212f084-45ab-4897-be30-2d33314f4aca.png)

Reply:
> According to the above statement, dependencyManagement in org.example:example-child:2.0.0 is the highest priority.


It has no logic. We should be able to change any child dependency in the base pom.
Perhaps you mixed up `example-child` and `base`? `project B` from maven docs is `base` in your example: `When maven is run on project B`. Or i missed something?

I reproduced your example.
Maven has the following priority:
1. com.example:base:4.0.0
2. com.example:parent:3.0.0
3. org.example:example-child:2.0.0
4. org.example:example-child-parent:2.0.0

also i checked Trivy with this PR. Trivi works the same way.

Reply:
Hmm. Even though `org.example:example-child:2.0.0` has `dependencyManagement` and `dependencies` both, Does `com.example:base:4.0.0` take precedence? In the above example, is `2.0.0` selected rather than `3.0.0`?

```
<dependencyManagement>
  <dependencies>
    <dependency>
      <groupId>example</groupId>
      <artifactId>jackson-databind</artifactId>
      <version>3.0.0</version>
    </dependency>
  </dependencies>
</dependencyManagement>

<dependencies>
  <dependency>
    <groupId>example</groupId>
    <artifactId>jackson-databind</artifactId>
  </dependency>
</dependencies>
```

Reply:
`dependencyManagement` from root `pom.xml` has highest priority.
e.g. we have:
`com.example:base:4.0.0`:
```
<dependencyManagement>
  <dependencies>
    <dependency>
      <groupId>example</groupId>
      <artifactId>jackson-databind</artifactId>
      <version>2.2.2</version>
    </dependency>
  </dependencies>
</dependencyManagement>

<dependencies>
  <dependency>
    <groupId>org.example</groupId>
    <artifactId>example-child</artifactId>
    <version>2.0.0</version>
  </dependency>
</dependencies>
```

and `org.example:example-child:2.0.0`:
```
<dependencyManagement>
  <dependencies>
    <dependency>
      <groupId>example</groupId>
      <artifactId>jackson-databind</artifactId>
      <version>3.3.3</version>
    </dependency>
  </dependencies>
</dependencyManagement>

<dependencies>
  <dependency>
    <groupId>example</groupId>
    <artifactId>jackson-databind</artifactId>
  </dependency>
</dependencies>
```

In this case version of `jackson-databind` is `2.2.2`.

Reply:
Got it. Thanks for clarifying.